app/src/main/kotlin/com/hedvig/app/service/push/RegisterPushTokenAuthEventListener.kt

```diff
@@ -0,0 +1,46 @@
+package com.hedvig.app.service.push
+
+import android.content.SharedPreferences
+import arrow.core.Either
+import arrow.core.continuations.either
+import com.apollographql.apollo3.ApolloClient
+import com.hedvig.android.apollo.OperationResult
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.auth.event.AuthEventListener
+import giraffe.NotificationRegisterDeviceMutation
+import slimber.log.e
+import slimber.log.i
+
+class RegisterPushTokenAuthEventListener(
+  private val apolloClient: ApolloClient,
```

Comment:
Getting a circular dependency here because OkHttp depends on `AuthTokenRefreshingInterceptor`. Not sure how we can solve that without making a larger refactoring, do you have any ideas @StylianosGakis?

Reply:
Gonna pick this up now and see what's up